src/libraries/Common/src/Interop/FreeBSD/Interop.Process.cs

```diff
@@ -76,22 +76,25 @@ internal static unsafe int[] ListAllPids()
         /// <param name="pid">The PID of the process</param>
         public static unsafe string? GetProcPath(int pid)
         {
-            Span<int> sysctlName = stackalloc int[4];
+            Span<int> sysctlName = stackalloc int[] { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, pid };
             byte* pBuffer = null;
             int bytesLength = 0;
 
-            sysctlName[0] = CTL_KERN;
-            sysctlName[1] = KERN_PROC;
-            sysctlName[2] = KERN_PROC_PATHNAME;
-            sysctlName[3] = pid;
+            try
+            {
+                Interop.Sys.Sysctl(sysctlName, ref pBuffer, ref bytesLength);
+                string path = System.Text.Encoding.UTF8.GetString(pBuffer, (int)bytesLength - 1);
+                Marshal.FreeHGlobal((IntPtr)pBuffer);
```

Comment:
Did you mean to delete this line?  Otherwise isn't this double-freeing?

Reply:
Yes, this was redundant. :)